src/main/kotlin/no/nav/familie/ef/sak/service/VedtaksbrevService.kt

```diff
@@ -45,15 +48,27 @@ class VedtaksbrevService(private val brevClient: BrevClient,
                                       brevRequest)
     }
 
-    fun lagreBrev(behandlingId: UUID) {
+    fun lagreBrevUtkast(behandlingId: UUID): Vedtaksbrev {
         val request = lagBrevRequest(behandlingId)
         val pdf = lagPdf(request)
         val brev = Vedtaksbrev(behandlingId, request, null, Fil(pdf), null)
-        brevRepository.insert(brev)
+        return brevRepository.insert(brev)
     }
 
-    fun forhåndsvisBrev(behandlingId: UUID): ByteArray{
-        return  lagPdf(lagBrevRequest(behandlingId))
+    fun lagreEndeligBrev(behandlingId: UUID): Vedtaksbrev {
+        try {
+            brevRepository.findById(behandlingId).get().let {
+                val endeligRequest = it.utkastBrevRequest.copy(signaturBeslutter = SikkerhetContext.hentSaksbehandlerNavn())
+                return brevRepository.update(it.copy(pdf = Fil(lagPdf(endeligRequest)), brevRequest = endeligRequest))
+            }
+        } catch (exception: Exception) {
+            error("Feil ved lagring av endelig brev. Finner ikke brev-utkast for behandling $behandlingId")
+            throw exception
+        }
```

Comment:
Her er det mulig å erstatte .findById og .get med `findByIdOrThrow` som gjør samme sak :) 
```suggestion
        val vedtaksbrev = brevRepository.findByIdOrThrow(behandlingId)
        val endeligRequest = vedtaksbrev.utkastBrevRequest.copy(signaturBeslutter = SikkerhetContext.hentSaksbehandlerNavn())
        return brevRepository.update(vedtaksbrev.copy(pdf = Fil(lagPdf(endeligRequest)), brevRequest = endeligRequest))
```

Reply:
Ah, nice 👏